pilot/pkg/serviceregistry/aggregate/controller.go

```diff
@@ -65,6 +65,17 @@ func (c *Controller) ServicesForWaypoint(key model.WaypointKey) []model.ServiceI
 	return res
 }
 
+func (c *Controller) ServicesWithWaypoint(key string) []model.ServiceWaypointInfo {
+	if !features.EnableAmbient || !features.EnableIngressWaypointRouting {
```

Comment:
nit: based on the name, i'd expect this function to work even if I wasn't calling it for ingress  purposes.